### PR TITLE
do not use server side apply for CRD

### LINKED ISSUE #28189: Upgrading from 1.7.0 to  1.8.0-alpha.0 failed
I am using canary upgrade according to official doc.

But it failed waiting for ingress gateway with this error 

```
2020-10-22T12:51:42.053822Z	warn	ads	ADS:LDS: ACK ERROR router~10.244.0.33~istio-ingressgateway-846f77dc5f-rbmnk.istio-system~istio-system.svc.cluster.local-1 Internal:Error adding/updating listener(s) 0.0.0.0_8080: Proto constraint validation failed (HttpConnectionManagerValidationError.HttpFilters[i]: ["embedded message failed validation"] | caused by HttpFilterValidationError.Name: ["value length must be at least " '\x01' " runes"]): stat_prefix: "outbound_0.0.0.0_8080"
rds {
  config_source {
    ads {
    }
    resource_api_version: V3
  }
  route_config_name: "http.80"
}
http_filters {
}
http_filters {
  name: "envoy.filters.http.cors"
  typed_config {
    [type.googleapis.com/envoy.extensions.filters.http.cors.v3.Cors] {
    }
  }
}
http_filters {
  name: "envoy.filters.http.fault"
  typed_config {
    [type.googleapis.com/envoy.extensions.filters.http.fault.v3.HTTPFault] {
    }
  }
}
http_filters {
}
http_filters {
  name: "envoy.filters.http.router"
  typed_config {
    [type.googleapis.com/envoy.extensions.filters.http.router.v3.Router] {
    }
  }
}
tracing {
  client_sampling {
    value: 100
  }
  random_sampling {
    value: 1
  }
  overall_sampling {
    value: 100
  }
}
http_protocol_options {
}
server_name: "istio-envoy"
access_log {
  name: "envoy.access_loggers.file"
  typed_config {
    [type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog] {
      path: "/dev/stdout"
      log_format {
        text_format: "[%START_TIME%] \"%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%\" %RESPONSE_CODE% %RESPONSE_FLAGS% \"%UPSTREAM_TRANSPORT_FAILURE_REASON%\" %BYTES_RECEIVED% %BYTES_SENT% %DURATION% %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% \"%REQ(X-FORWARDED-FOR)%\" \"%REQ(USER-AGENT)%\" \"%REQ(X-REQUEST-ID)%\" \"%REQ(:AUTHORITY)%\" \"%UPSTREAM_HOST%\" %UPSTREAM_CLUSTER% %UPSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_REMOTE_ADDRESS% %REQUESTED_SERVER_NAME% %ROUTE_NAME%\n"
      }
    }
  }
}
use_remote_address {
  value: true
}
forward_client_cert_details: SANITIZE_SET
set_current_client_cert_details {
  subject {
    value: true
  }
  cert: true
  dns: true
  uri: true
}
upgrade_configs {
  upgrade_type: "websocket"
}
stream_idle_timeout {
}
normalize_path {
  value: true
}

```

I have httpbin deployed before and with `samples/httpbin/httpbin-gateway.yaml` applied.

### PR DESCRIPTION
**Pull request by richardwxn**
resolve https://github.com/istio/istio/issues/28189

not clear what is a better way to add test for it as it happens during upgrade, so would be better to keep monitoring the daily upgrade test

### CODE REVIEW

**Comment by howardjohn** [comment]
Can we open a k8s bug for this?

File: operator/pkg/helmreconciler/apply.go [hunk 1]
```diff
@@ -193,8 +193,7 @@ func (h *HelmReconciler) ApplyObject(obj *unstructured.Unstructured, serverSideA
 	}
 
 	gvk := obj.GetObjectKind().GroupVersionKind()
-
-	if serverSideApply {
+	if serverSideApply && obj.GetKind() != name.CRDStr {
```
Comment (howardjohn) [inline operator/pkg/helmreconciler/apply.go:196]: Can you add a comment with TODO linking to issue?

Comment (richardwxn) [reply operator/pkg/helmreconciler/apply.go:197]: added

**Comment by richardwxn** [comment]
added bug link in the comment

**Review by howardjohn — APPROVED** [review]

**Pull request merged**